javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistEnumDeclaration.java

```diff
@@ -206,10 +205,10 @@ public String getName() {
         try {
             CtClass superClass = ctClass.getSuperclass();
             if (superClass != null) {
-                SymbolReference<ResolvedMethodDeclaration> ref = new JavassistClassDeclaration(superClass, typeSolver).solveMethod(name, argumentsTypes, staticOnly);
-                if (ref.isSolved()) {
-                    candidates.add(ref.getCorrespondingDeclaration());
-                }
+                new JavassistClassDeclaration(superClass, typeSolver)
+                        .solveMethod(name, argumentsTypes, staticOnly)
+                        .getCorrespondingDeclaration()
+                        .ifPresent(candidates::add);
```

Comment:
worth extracting out a method here for the shared behaviour in the try blocks?

again, just thinking aloud - can be a separate issue/PR to keep this one simple / focussed just on the introduction of optional

Reply:
(see also JavassistInterfaceDeclaration.java -- might be more hassle than it's worth though!)

Reply:
Maybe we could make it in another PR to keep this one focused only on implementing Optional. What do you think?